scripts/saveState.js

```diff
@@ -46,16 +37,19 @@ function addToSandwich(type, item){
 	
 	if(!type.localeCompare("subgoal")){ 		
 		var subArr = getSubgoalArrayFromLocal();
-		drawSubgoal(item.id);
+		//drawSubgoal(item.id);
```

Comment:
Calling drawSubgoal in the beginning of addToSandwich creates a problem with this call stack: drawSubgoal --> editSubgoal --> saveSubgoal --> addToSandwich --> drawSubgoal. The state of the subgoal and/or some other state variables have changed between the first and last drawSubgoal, so the wrong information gets stored and displayed. It shouldn't affect actions since the drawSubgoal call occurs outside of those if blocks.